pumps/mongo_aggregate.go

```diff
@@ -261,12 +288,30 @@ func (m *MongoAggregatePump) WriteData(ctx context.Context, data []interface{})
 		m.WriteData(ctx, data)
 	} else {
 		// calculate aggregates
-		analyticsPerOrg := analytics.AggregateData(data, m.dbConf.TrackAllPaths, m.dbConf.IgnoreTagPrefixList, m.dbConf.StoreAnalyticsPerMinute, true)
-
+		analyticsPerOrg := analytics.AggregateData(data, m.dbConf.TrackAllPaths, m.dbConf.IgnoreTagPrefixList, m.dbConf.MongoURL, m.dbConf.AggregationTime, true)
 		// put aggregated data into MongoDB
 		for orgID, filteredData := range analyticsPerOrg {
 			err := m.DoAggregatedWriting(ctx, orgID, filteredData)
 			if err != nil {
+				// checking if the error is related to the document size and AggregateSelfHealing is enabled
+				if strings.Contains(err.Error(), "Size must be between 0 and 16793600(16MB)") && m.dbConf.EnableAggregateSelfHealing {
+					// if the AggregationTime setting is already set to 1, we can't do anything else
+					if m.dbConf.AggregationTime == 1 {
+						m.log.Warning("AggregationTime is equal to 1 minute, unable to reduce it further. Skipping self-healing.")
+						return err
+					}
```

Comment:
Even if the value is set to 1, shouldn't we create a new document?
We cannot divide the aggregationTime, but makes sense to create a new document anyway?

Reply:
I don't agree with this one @sredxny - If we are in the minimum aggregation time, trying to write records and still hitting the max MB, I think we should error

Reply:
Hmm. If the AggregationTime is already set to 1, it means that we're creating new documents every 1 minute. So if you've created a document at 10:46 and it hits 16mb before 10:47, we can't create a new document, because the time is still 10:46, right? If we want to handle these situations, we should include seconds (i.e: creating documents every 30 seconds).
If the document hits 16mb, but now the time is 10:47, there won't be any problem because a new document will be created automatically.
Let me know if there is something I am not seeing!